### PR TITLE
Introduces Rubocop

### PR DESCRIPTION
**Pull request by hennevogel**

### CODE REVIEW

**Comment by hennevogel** [comment]
No opinion from anyone in the @openSUSE/open-build-service team?

**Comment by digitaltom** [comment]
In the CSM workshop we agreed to share a common SUSE styleguide: https://github.com/SUSE/style-guides/blob/master/rubocop-suse.yml 
The different teams can add an overlay config to this default to cover individual needs of their projects.

**Comment by adrianschroeter** [comment]
Didn't we agreed to speak about it during refactoring session?

For my side, +1 for code length check, I see not the benefit of the whitespace check, just more work there. But no strong opinion.

**Comment by ChrisBr** [comment]
I'd definitely avoid using 

```
# rubocop:disable Metrics/LineLength
```
and rather fix the "root problem".

When we agree to introduce rubucop then we should avoid exception as far as possible ...

File: src/api/.rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+inherit_from: .rubocop_todo.yml
+AllCops:
+  RunRailsCops: true
+  Exclude:
+    - 'tmp/**/*'
+
+# Configuration parameters: AllowURI, URISchemes.
+Metrics/LineLength:
+  Max: 150
+  Exclude:
```
Comment (ChrisBr) [inline src/api/.rubocop.yml]: I'd add 
```
  # To make it possible to copy or click on URIs in the code, we allow lines
  # contaning a URI to be longer than Max.
  AllowURI: true
  URISchemes:
    - http
    - https
```

And I think 150 char are too much ...

Comment (bgeuken) [inline src/api/.rubocop.yml]: +1

I would prefer something between 80 and 100. With a tendency towards 80;-)

**Comment by hennevogel** [comment]
@digitaltom did you also discuss how to pull this file in? What happens if this is updated, do I need to watch it manually?

**Comment by hennevogel** [comment]
Alright @openSUSE/open-build-service people, what do we do with this one?

**Comment by hennevogel** [comment]
Had to manually rebase in a branch. This is now in #1008

**Pull request closed without merging**